crates/api_models/src/subscription.rs

```diff
@@ -0,0 +1,151 @@
+use common_utils::{events::ApiEventMetric, pii};
+use masking::Secret;
+use utoipa::ToSchema;
+
+use crate::{
+    customers::{CustomerRequest, CustomerResponse},
+    payments::CustomerDetailsResponse,
+};
+
+/// Request payload for creating a subscription.
+///
+/// This struct captures details required to create a subscription,
+/// including plan, profile, merchant connector, and optional customer info.
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
+pub struct CreateSubscriptionRequest {
+    /// Unique identifier for the subscription (optional, generated if missing).
+    pub subscription_id: Option<common_utils::id_type::SubscriptionId>,
+
+    /// Associated profile ID for this subscription.
+    pub profile_id: common_utils::id_type::ProfileId,
+
+    /// Identifier for the subscription plan.
+    pub plan_id: Option<String>,
+
+    /// Optional coupon code applied to the subscription.
+    pub coupon_code: Option<String>,
+
+    /// Optional merchant connector account ID for routing payments.
+    pub merchant_connector_account_id: Option<common_utils::id_type::MerchantConnectorAccountId>,
+
+    /// Whether to immediately confirm the subscription on creation.
+    pub confirm: Option<bool>,
+
+    /// Optional customer ID associated with this subscription.
+    pub customer_id: Option<common_utils::id_type::CustomerId>,
+
+    /// Optional full customer request data.
+    pub customer: Option<CustomerRequest>,
+}
+
+impl CreateSubscriptionRequest {
+    /// Retrieves the `customer_id` either from the top-level field
+    /// or from the nested [`CustomerRequest`].
+    pub fn get_customer_id(&self) -> Option<&common_utils::id_type::CustomerId> {
+        self.customer_id
+            .as_ref()
+            .or_else(|| self.customer.as_ref()?.customer_id.as_ref())
+    }
+}
+
+/// Response payload returned after successfully creating a subscription.
+///
+/// Includes details such as subscription ID, status, plan, merchant, and customer info.
+#[derive(Debug, Clone, serde::Serialize, ToSchema)]
+pub struct CreateSubscriptionResponse {
+    /// Unique identifier for the subscription.
+    pub subscription_id: common_utils::id_type::SubscriptionId,
```

Comment:
```suggestion
    pub id: common_utils::id_type::SubscriptionId,
```

resource api name convention